### PR TITLE
redocly docs generation

### LINKED ISSUE #7340: [Docs] Replace Meshery API doc generation with ReDocly
#### Current State
Currently, API docs, for Meshery are generated using Swagger.

#### Desired State
1. Move the API docs generation process with [ReDocly](https://redocly.com/docs/cli/installation/).
2. A GitHub workflow which automates doc generation (updating release version, bundling.....)
Deployment guide:  https://redocly.com/docs/redoc/deployment/html/ 

Reference OpenAPI spec: https://github.com/meshery/meshery/blob/6ab5abf1118a190b80cd4fb7226b2b20bb165a29/docs/_data/swagger.yml

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Freedisch**
**Notes for Reviewers**

This PR fixes #7340 
I moved the api docs generation to redocly in the make file and added a workflow which automates doc generation based on https://redocly.com/docs/redoc/deployment/cli/

Screenshot 2023-04-12 191127

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@senali-d @vishalvivekm

**Comment by leecalcote** [comment]
@Freedisch thank you for this. We've begun using redocly for Meshery Cloud's API docs and have run into a few concerns:

1. theme coloring doesn't seem to work. Perhaps, we're doing it wrong.
1. are we able to sort the APIs alphabetically?
1. templating doesn't seem to work. For example, incorporating release version number takes a bit more work to be done in another workflow step. Unless, the redocly makes this possible with templating.

I wonder if you might be able to overcome these items.

**Comment by leecalcote** [comment]
@Freedisch There is a bi-weekly Meshery CI meeting tomorrow. // @Revolyssup

**Comment by leecalcote** [comment]
T-bo, this is good news. I encourage you to put in agenda item on today’s CA meeting.  

@Revolyssup should be reaching out to you to ensure that this PR has a spot on the agenda - whether you can make the meeting or not.

**Comment by leecalcote** [comment]
@Freedisch, T-bo, will you drop a copy of the current state of the generated html file here?

**Comment by Freedisch** [comment]
> @Freedisch, T-bo, will you drop a copy of the current state of the generated html file here?

Sure here it is https://drive.google.com/file/d/1xBL-i6-zIwDazbjF1CpocxkfqYwfvCpC/view?usp=sharing

File: Makefile [hunk 1]
```diff
@@ -314,9 +314,8 @@ swagger: swagger-build
 	swagger serve ./server/helpers/swagger.yaml
 
 ## Build Meshery REST API documentation
-swagger-docs-build:
-	swagger generate spec -o ./docs/_data/swagger.yml --scan-models; \
-	swagger flatten ./docs/_data/swagger.yml -o ./docs/_data/swagger.yml --with-expand --format=yaml
+redocly-docs-build:
+	npx @redocly/cli build-docs ./docs/_data/swagger.yml
```
[the remark this answers is not in the document]
Comment (Freedisch) [reply Makefile:318]: I noticed right after my PR, my next commit will have it back.

**Comment by Freedisch** [comment]
Added a config file that set the different options and theme while building the docs (that file needs to be at root of the project for it to work https://redocly.com/docs/cli/configuration/

**Comment by Freedisch** [comment]
> @Freedisch thank you for this. We've begun using redocly for Meshery Cloud's API docs and have run into a few concerns:
> 
> 1. theme coloring doesn't seem to work. Perhaps, we're doing it wrong.
> 2. are we able to sort the APIs alphabetically?
> 3. templating doesn't seem to work. For example, incorporating release version number takes a bit more work to be done in another workflow step. Unless, the redocly makes this possible with templating.
> 
> I wonder if you might be able to overcome these items.

Regarding the templating, I haven't found something allowing to do it in redocly

**Review by MUzairS15 — COMMENTED** [review]
@Freedisch How are you ensuring the docs generated by Redocly is referenced at `RestAPI Refernece` page.
Right now in this PR still swagger docs are present.

**Comment by Freedisch** [comment]
> @Freedisch How are you ensuring the docs generated by Redocly is referenced at `RestAPI Refernece` page. Right now in this PR still swagger docs are present.

Currently this PR doesn't include a modification of the rest api reference page to redocly but I think this can be achieved by replacing the url of swagger documentation to redocly. But I still need to find that path or url the swagger docs.

**Comment by Chadha93** [comment]
@Freedisch  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Review by leecalcote — APPROVED** [review]
Merging, so that we can press forward with exploration of use. This PR doesn't not remove the current documentation system, but adds an alternative.

**Pull request merged**